modules/bitgo/test/v2/unit/wallet.ts

```
@@ -5100,4 +5379,349 @@ describe('V2 Wallet:', function () {
+            memo: {
+              type: 'type',
+              value: 'test memo',
+            },
+          },
+          feeInfo: {
+            fee: 5000,
+            feeString: '5000',
+          },
+        });
+      });
+
+      it('should handle solInstruction with pending approval ID', async function () {
+        const prebuildTxWithIntent = sandbox.stub(TssUtils.prototype, 'prebuildTxWithIntent');
+        prebuildTxWithIntent.resolves({ ...txRequest, state: 'pendingApproval', pendingApprovalId: 'some-id' });
+        prebuildTxWithIntent.calledOnceWithExactly({
+          reqId,
+          intentType: 'solInstruction',
+          solInstructions: [testCustomInstruction],
+          recipients: testRecipients,
+        });
+
+        const txPrebuild = await custodialTssSolWallet.prebuildTransaction({
+          reqId,
+          type: 'solInstruction',
+          solInstructions: [testCustomInstruction],
+          recipients: testRecipients,
+        });
+
+        txPrebuild.should.deepEqual({
+          walletId: custodialTssSolWallet.id(),
+          wallet: custodialTssSolWallet,
+          txRequestId: 'id',
+          txHex: 'ababcdcd',
+          pendingApprovalId: 'some-id',
+          buildParams: {
+            type: 'solInstruction',
+            solInstructions: [testCustomInstruction],
+            recipients: testRecipients,
+          },
+          feeInfo: {
+            fee: 5000,
+            feeString: '5000',
+          },
+        });
+      });
+
+      it('should handle empty solInstructions array', async function () {
+        const prebuildTxWithIntent = sandbox.stub(TssUtils.prototype, 'prebuildTxWithIntent');
+        prebuildTxWithIntent.resolves(txRequest);
+        prebuildTxWithIntent.calledOnceWithExactly({
+          reqId,
+          intentType: 'solInstruction',
+          solInstructions: [],
+          recipients: testRecipients,
+        });
+
+        const txPrebuild = await tssSolWallet.prebuildTransaction({
```

We should throw an error for this case?

---

Yeah, good catch. Added an assertion check